bentoml/_internal/service/inference_api.py

```diff
@@ -39,7 +45,77 @@ def __init__(
         InferenceAPI._validate_route(route)
 
         self.name = name
+        self.needs_ctx = False
+        self.ctx_param = None
         self.func = user_defined_callback
+        input_type = input_descriptor.input_type()
+        self.multi_input = isinstance(input_type, dict)
```

Comment:
should we do `isinstance(input_descriptor, MultiInput)` here just in case another new IODescriptor added that has input type dict?

Reply:
This is checking that the `input_type` is a `dict` (and therefore not a `t.Type[dict]`). It's a bit weird but I decided the way that we'd represent multiple outputs is a dict of types, while an IO descriptor that returns a dict would just return `dict` itself.

Reply:
Wouldn't `isinstance(input_descriptor, MultiInput)` be a more direct indication?

Reply:
I slightly prefer to ignore type check here and use `isinstance(input_descriptor, MultiInput)` instead, as it makes the code more readable

Reply:
This was a conscious choice to allow users to write their own I/O descriptor that passes kwargs.